pkg/controller/alicloudmachineclass.go

```diff
@@ -193,7 +194,7 @@ func (c *controller) reconcileClusterAlicloudMachineClass(class *v1alpha1.Aliclo
 		if c.deleteMigratedMachineClass && annotationPresent && len(machines) == 0 {
 			// If controller has deleteMigratedMachineClass flag set
 			// and the migratedMachineClass annotation is set
-			err = c.controlMachineClient.AlicloudMachineClasses(class.Namespace).Delete(class.Name, &metav1.DeleteOptions{})
+			err = c.controlMachineClient.AlicloudMachineClasses(class.Namespace).Delete(context.TODO(), class.Name, metav1.DeleteOptions{})
```

Comment:
You will need to determine the right way to pass the context here. I think `context.TODO()` is a hack. Ideally we will need some sort of timeout. However, probably need a global context as well from which we derive contexts. 

You will need to research more about it. Something [like this](https://github.com/gardener/gardener/blob/77e09dd522d7bfb682cc35d0be437201b606061b/pkg/utils/kubernetes/shoot.go#L47-L50)?